### PR TITLE
aws-s3 input default content-type

### LINKED ISSUE #25697: S3 input: Add fallback configuration for content type so that events can be decoded
If the S3 object has a `Content-Type` of `application/json`, we will decode it as JSON in the input.  Sometimes, the uploaded S3 objects do not have content type set.  For these cases, it will be helpful to have an optional configuration to tell the S3 input to fallback to a specified content type of choice when no Content-Type is specified so that we will still attempt to decode them.

@andrewkroh

### PR DESCRIPTION
**Pull request by leehinman**
## What does this PR do?

Adds a new option `content_type` to the aws-s3 input.  This allows you
to override the Content-Type that was given to the S3 object when it
was uploaded.

## Why is it important?

Content-Type is used to determine if the S3 object should be treated
as JSON or not.  If the object was uploaded with the incorrect
Content-Type that can prevent us from parsing the file correctly.

## Checklist

- [x] My code follows the style guidelines of this project
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
- [x] I have made corresponding changes to the documentation
~~- [ ] I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

Setup SQS and S3 bucket and download files.

## Related issues

- Closes #25697

### CODE REVIEW

File: x-pack/filebeat/docs/inputs/input-aws-s3.asciidoc [hunk 1]
```diff
@@ -49,6 +49,14 @@ seconds. The maximum is half of the visibility timeout value.
 The size in bytes of the buffer that each harvester uses when fetching a file.
 This only applies to non-JSON logs. The default is `16 KiB`.
 
+[id="input-{type}-content_type"]
+[float]
+==== `content_type`
+
+A standard MIME type describing the format of the object data.  This
+can be set to override the MIME type that was given to the object when
+it was uploaded. example: `application/json`
```
Comment (kaiyan-sheng) [inline x-pack/filebeat/docs/inputs/input-aws-s3.asciidoc:58]:
```suggestion
it was uploaded. For example: `application/json`.
```

**Review by kaiyan-sheng — APPROVED** [review]
Looks good to me.

File: x-pack/filebeat/input/awss3/config.go [hunk 2]
```diff
@@ -82,6 +83,9 @@ func (f *readerConfig) Validate() error {
 	if f.MaxBytes <= 0 {
 		return fmt.Errorf("max_bytes <%v> must be greater than 0", f.MaxBytes)
 	}
+	if f.ExpandEventListFromField != "" && f.ContentType != "" && f.ContentType != "application/json" {
+		return fmt.Errorf("content_type must be `application/json` to be used with expand_event_list_from_field")
```
Comment (andrewkroh) [inline x-pack/filebeat/input/awss3/config.go:87]:
```suggestion
		return fmt.Errorf("content_type must be `application/json` when expand_event_list_from_field is used")
```

Just a suggestion

**Review by andrewkroh — APPROVED** [review]
LGTM.

**Pull request merged**